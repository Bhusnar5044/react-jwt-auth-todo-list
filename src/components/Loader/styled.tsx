import { keyframes, styled } from '@utils';

const logoSpin = keyframes`
   from {
            transform: rotate(0deg);
        }
        to {
            transform: rotate(360deg);
        }
`;
export const StyledLoader = styled.img`
    width: 10rem;
    height: auto;
    margin: auto;
    animation: ${logoSpin} infinite 20s linear;
`;
