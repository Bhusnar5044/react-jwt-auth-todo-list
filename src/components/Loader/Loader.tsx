import { FC, memo } from 'react';
import { StyledLoader } from './styled';

export const Loader: FC = memo(() => {
    return <StyledLoader src="/reactLogo.svg" title="loader" />;
});

Loader.displayName = 'Loader';
