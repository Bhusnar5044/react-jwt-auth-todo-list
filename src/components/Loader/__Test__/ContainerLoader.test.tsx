import { render, screen } from '@test-utils';
import { ContainerLoader } from '../Loader';

describe('ContainerLoader', () => {
    it('should render properly', () => {
        const { container } = render(<ContainerLoader />);
        expect(container).toMatchSnapshot();
        expect(screen.getByTitle('loader')).toBeInTheDocument();
    });
});
