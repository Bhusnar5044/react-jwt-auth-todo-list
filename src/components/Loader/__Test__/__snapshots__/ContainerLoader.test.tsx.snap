// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerLoader should render properly 1`] = `
.c0 {
  width: 10rem;
  height: auto;
  margin: auto;
  animation: gOaCMp infinite 20s linear;
}

<div>
  <img
    class="c0"
    src="/reactLogo.svg"
    title="loader"
  />
</div>
`;
